x-pack/solutions/security/plugins/security_solution/public/data_view_manager/redux/slices.test.ts

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DataView } from '@kbn/data-views-plugin/common';
+import {
+  sharedDataViewManagerSlice,
+  createDataViewSelectionSlice,
+  initialSharedState,
+  initialScopeState,
+} from './slices';
+import { selectDataViewAsync } from './actions';
+import { DataViewManagerScopeName } from '../constants';
+
+describe('slices', () => {
+  describe('dataViewSelectionSlice', () => {});
+
+  describe('sharedDataViewManagerSlice', () => {
+    const { reducer, actions } = sharedDataViewManagerSlice;
+
+    describe('setDataViews', () => {
+      it('should set dataViews and change status to ready', () => {
+        const dataViews = [{ id: '1', title: 'test view' }];
+
+        const state = reducer(initialSharedState, actions.setDataViews(dataViews));
+
+        expect(state.dataViews).toEqual(dataViews);
+        expect(state.status).toBe('ready');
+      });
+    });
+
+    describe('updateDataView', () => {
+      it('should update a persisted data view', () => {
+        const initialState = {
+          ...initialSharedState,
+          dataViews: [{ id: '1', title: 'test view' }],
+        };
+
+        const updatedDataView = {
+          id: '1',
+          title: 'updated view',
+          isPersisted: () => true,
+          toSpec: () => ({ id: '1', title: 'updated view' }),
+        } as unknown as DataView;
+
+        const state = reducer(initialState, actions.updateDataView(updatedDataView));
+
+        expect(state.dataViews[0]).toEqual({ id: '1', title: 'updated view' });
+      });
+
+      it('should update an ad hoc data view', () => {
+        const initialState = {
+          ...initialSharedState,
+          adhocDataViews: [{ title: 'ad hoc view' }],
+        };
+
+        const updatedDataView = {
```

Comment:
This test isn't showing an update as nothing changes in the actual spec. Do you want to add a `title` value here?